### PR TITLE
Feature/cpp rbk

### PR DESCRIPTION
**Pull request by thibaudteil**
* **Tickets addressed:** bsk-773 and 774
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
This PR contains the work for two tickets and revolves around converting the RigidBodyKinematics C library to a Cpp/Eigen library. The second ticket it address focuses around adding tests for all these functions with google tests.

- The first commit adds the hpp file containing the function names and type definitions
- The second commit populates the function definitions
- The third commit adds the google tests (and modifies the Cmake the find these tests)
- The final commit wraps the new functions in swig

## Verification
All functions converted were tested against properties of rotations and against the previous implementation (which has heritage)

## Documentation
Doc strings were added for each function

## Future work
None

### CODE REVIEW

File: src/architecture/utilities/rigidBodyKinematics.hpp
Comment (patkenneally) [inline src/architecture/utilities/rigidBodyKinematics.hpp]: :large_orange_diamond: We will have this automated soon with pre-commit. If that comes before this then all good, if not, when parameters run beyond the line length we should choose to drop each one to a new line. E.g (ignoring my jank indenting). 
```
Eigen::Vector3d ddmrp(const Eigen::Vector3d& mrp, 
                                       const Eigen::Vector3d& dmrp,
                                       const Eigen::Vector3d& omega, 
                                       const Eigen::Vector3d& domega);
```

**Review by patkenneally — APPROVED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/architecture/utilities/rigidBodyKinematics.hpp
Comment (thibaudteil) [reply src/architecture/utilities/rigidBodyKinematics.hpp]: will do it now!

**Review by patkenneally — APPROVED** [review]

**Pull request merged**
